src/app/qgsmergeattributesdialog.cpp

```diff
@@ -183,12 +186,17 @@ void QgsMergeAttributesDialog::createTableWidgetContents()
 
     QComboBox *cb = createMergeComboBox( mFields.at( idx ).type(), col );
     if ( ! mVectorLayer->dataProvider()->pkAttributeIndexes().contains( mFields.fieldOriginIndex( idx ) ) &&
-         mFields.at( idx ).constraints().constraints() & QgsFieldConstraints::ConstraintUnique )
+         mFields.at( idx ).constraints().constraints() & QgsFieldConstraints::ConstraintUnique || mHiddenAttributes.contains( idx ) )
     {
       cb->setCurrentIndex( cb->findData( "skip" ) );
     }
     mTableWidget->setCellWidget( 0, col, cb );
 
+    if ( setup.type() == QLatin1String( "Hidden" ) )
```

Comment:
Here the same method as in line 189 can be used to detect hidden field:
```suggestion
    if ( mHiddenAttributes.contains( idx ) )
```

Reply:
Thanks for the suggestions! Now both of them applied to the commit